src/simcore_service_deployment_agent/git_url_watcher.py

```diff
@@ -120,6 +120,28 @@ async def _git_fetch(directory: str):
     await run_cmd_line(cmd, f"{directory}")
 
 
+async def _git_get_latest_matching_tag_capture_groups(
+    directory: str, regexp: str
+) -> Optional[str]:  # pylint: disable=unsubscriptable-object
```

Comment:
why is this pylint disable comment here?

Reply:
and by the way, groups() returns a list or a string?

Reply:
A more generic question: is there a reason why we avoid using a python client for git instead of the CLI? e.g  https://gitpython.readthedocs.io/en/stable/tutorial.html ?